### PR TITLE
prelude: add protocol information through JSON

### PR DESCRIPTION
**Pull request by ToToL**
Prelude is based on what can be found in packet but now, Suricata provide more information in JSON format for HTTP, TLS, SSH and SMTP. This PR add the support for prelude to read these JSON and integrate them into IDMEF alerts.

### CODE REVIEW

**Comment by ToToL** [comment]
up

File: src/alert-prelude.c [hunk 1]
```diff
@@ -478,6 +575,157 @@ static int PacketToDataV6(const Packet *p, const PacketAlert *pa, idmef_alert_t
     SCReturnInt(0);
 }
 
+/**
+ * \brief Convert JSON object to Prelude additional data with
+ * the right type of data.
+ *
+ * \return 0 if ok
+ */
+int JsonToAdditionnalData(const char * key, json_t * value, idmef_alert_t *alert ) {
```
Comment (inliniac) [inline src/alert-prelude.c]: this use of json_t without HAVE_JANSSON ifdefs looks suspicious to me. Will this code compile if libjansson is not available?

**Comment by ToToL** [comment]
I create a new PR with the new patch : #2869

**Comment by inliniac** [comment]
Replaced by #2869

**Pull request closed without merging**